addons/udes_priorities/models/priority.py

```diff
@@ -87,42 +105,33 @@ def _check_for_duplicate_names_for_same_pick_type(self):
             raise UserError("Name must be unique for the picking types it is shown on")
 
     @api.multi
-    def _number_of_oustanding_pickings(self):
-        Picking = self.env["stock.picking"]
-        self.ensure_one()
-        return Picking.search_count(
-            [("priority", "=", self.reference), ("state", "not in", ("cancel", "done")),]
-        )
-
-    @api.multi
-    def _check_for_oustanding_pickings(self):
+    def _check_for_outstanding_pickings(self):
 
         install_mode = self.env.context.get("install_mode", False)
 
         if install_mode:
             return
 
-        priorities_with_outstanding_picks = self.browse()
-        for priority in self:
-            if priority._number_of_oustanding_pickings() > 0:
-                priorities_with_outstanding_picks |= priority
+        privileged_user = (
+            self.env.uid == SUPERUSER_ID
+            or self.env.user.has_group("udes_security.group_debug_user")
+            or self.env.user.has_group("udes_security.group_trusted_user")
+        )
 
-        if priorities_with_outstanding_picks:
+        if self.affected_pickings > 0 and not privileged_user:
```

Comment:
Typo: outstanding_pickings instead of affected_pickings

Reply:
I think we still need `priorities_with_outstanding_picks`, since `_check_for_outstanding_pickings()` is called from write, which can contain more than one record in self. Although we can simplify the computation to something like:
```priorities_with_outstanding_picks = self.filtered(lambda p: p.outstanding_pickings > 0```

Reply:
I substituted "priorities_with_outstanding_picks" with "self.affected_pickings > 0" which is basically still checking for outstanding pickings but instead of checking all priorities it only checks the current one that is being visualised/modified which to me makes more sense since the changes to a priority would only affect pickings with that priority assigned. Is there a reason why we want to check all pickings regardless of the priority? Just trying to be specific and clear about the goal.

Reply:
This function is called from write() which is used to update multiple records, in that case self could contain more than one record and sefl.XXXX will fail for the same reason I have explained above.
Although we are updating only one priority using the UI we could have some code that calls write() and will trigger the error.